core/services/relay/evm/mercury_config_provider.go

```diff
@@ -0,0 +1,45 @@
+package evm
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	pkgerrors "github.com/pkg/errors"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/legacyevm"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+func newMercuryConfigProvider(lggr logger.Logger, chain legacyevm.Chain, opts *types.RelayOpts) (commontypes.ConfigProvider, error) {
+	if !common.IsHexAddress(opts.ContractID) {
+		return nil, pkgerrors.Errorf("invalid contractID, expected hex address")
```

Comment:
can use the std lib "errors" package and and below for such simple uses. pkgerrors is depreciated. best to avoid new uses.